### PR TITLE
ROX-18473: Add search input to listening endpoints table.

### PR DESCRIPTION
**Pull request by dvail**
## Description

As titled again.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed

Visit the listening endpoints page:
image

Enter a search value and press enter -or- click the `->` button:
image

Click the `X` button to clear the search:
image

Open multiple deployments and view the alignment of the nested tables:
image

### CODE REVIEW

**Comment by dvail** [comment]
Current dependencies on/for this PR:
* master
  * **PR #7021** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/7021" target="_blank">Graphite</a> 
    * **PR #7024** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/7024" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/7024?utm_source=stack-comment).

File: ui/apps/platform/src/Containers/Audit/ListeningEndpoints/ListeningEndpointsTable.tsx
Comment (dvail) [inline ui/apps/platform/src/Containers/Audit/ListeningEndpoints/ListeningEndpointsTable.tsx:1]: Two unrelated changes in this file:
1. Swap an empty Th for an empty Td in the header to avoid aXe devtool errors
2. Give the columns a consistent percentage width to avoid "lane change" UI

File: ui/apps/platform/src/Containers/Audit/ListeningEndpoints/ListeningEndpointsPage.tsx [hunk 1]
```diff
@@ -42,6 +59,19 @@ function ListeningEndpointsPage() {
             <PageSection isFilled className="pf-u-display-flex pf-u-flex-direction-column">
                 <Toolbar>
                     <ToolbarContent>
+                        <ToolbarItem variant="search-filter" className="pf-u-flex-grow-1">
+                            <SearchInput
+                                aria-label="Search by deployment"
+                                placeholder="Search by deployment"
+                                value={searchValue}
+                                onChange={onSearchInputChange}
+                                onSearch={() => setSearchFilter({ Deployment: searchValue })}
```
Comment (sachaudh) [inline ui/apps/platform/src/Containers/Audit/ListeningEndpoints/ListeningEndpointsPage.tsx:68]: Is it worth considering debouncing for the search input?

**Review by sachaudh — APPROVED** [review]
LGTM 👍🏼

File: ui/apps/platform/src/Containers/Audit/ListeningEndpoints/ListeningEndpointsPage.tsx [hunk 2]
```diff
@@ -30,7 +32,22 @@ const sortOptions = {
...
     const { sortOption, getSortParams } = useURLSort(sortOptions);
-    const { data, error, loading } = useDeploymentListeningEndpoints(sortOption, page, perPage);
+    const { searchFilter, setSearchFilter } = useURLSearch();
+    const [searchValue, setSearchValue] = useState(() => {
+        const filter = searchFilter.Deployment;
+        return Array.isArray(filter) ? filter.join(',') : filter;
+    });
+
+    const { data, error, loading } = useDeploymentListeningEndpoints(
+        searchFilter,
```
Comment (pedrottimark) [inline ui/apps/platform/src/Containers/Audit/ListeningEndpoints/ListeningEndpointsPage.tsx:42]: Hard to decide where to comment on unchanged code.

According to the pictures, pagination is still 1 - 10 after **central** matches only 2 deployments.

File: ui/apps/platform/src/Containers/Audit/ListeningEndpoints/ListeningEndpointsTable.tsx
Comment (pedrottimark) [reply ui/apps/platform/src/Containers/Audit/ListeningEndpoints/ListeningEndpointsTable.tsx:1]: Although off the main objective, can you include a picture that illustrates the improvement?

**Review by pedrottimark — APPROVED** [review]

File: ui/apps/platform/src/Containers/Audit/ListeningEndpoints/ListeningEndpointsPage.tsx (hunk 2 above)
Comment (pedrottimark) [reply ui/apps/platform/src/Containers/Audit/ListeningEndpoints/ListeningEndpointsPage.tsx:42]: Is there an edge case when data consists of less than the page size?
* when only one page of results
* for last page of results

Comment (dvail) [reply ui/apps/platform/src/Containers/Audit/ListeningEndpoints/ListeningEndpointsPage.tsx:42]: Yes this is an edge case caused by the indeterminate Pagination component variant due to the fact the we are not passing the total deployment count to the component.

At quick glance, it doesn't appear that there is a combination of props that will fix this issue. I'll make a note and investigate deeper, and if there isn't a good workaround we can revert to the typical pagination pattern we use.
